### PR TITLE
Added fastroping sounds

### PR DESCRIPTION
**Pull request by nomisum**
**When merged this pull request will:**
- Add Fastroping Sounds OGG (Roping + Thud)
- Add Fastroping Sounds License.MD
- Add Fastroping Sounds Code Proposition
- Work on #3475

### CODE REVIEW

File: addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf [hunk 1]
```diff
@@ -31,10 +31,19 @@ if (animationState _unit != "ACE_FastRoping") exitWith {
     [_unit, "ACE_FastRoping", 2] call EFUNC(common,doAnimation);
 };
 
+[_unit] spawn {
```
Comment (jokoho48) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: Spawn!!! please use PFH or waitAndExecute or waitUntilAndExecute

File: addons/fastroping/CfgSounds.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CfgSounds {
+    class ACE_Fastroping_Rope {
+        name = "ACE_Fastroping_Rope";
+        sound[] = {PATHTOF(data\sounds\fastroping_rope.ogg), db0, 1.0};
```
Comment (jonpas) [inline addons/fastroping/CfgSounds.hpp]: Missing quotes (for compatibility with other non-standard tools).
`QUOTE(PATHTOF(path))`

File: addons/fastroping/CfgSounds.hpp [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+        titles[] = {};
+    };
+    class ACE_Fastroping_Thud {
+        name = "ACE_Fastroping_Thud";
+        sound[] = {PATHTOF(data\sounds\fastroping_thud.ogg), db0, 1.0};
```
Comment (jonpas) [inline addons/fastroping/CfgSounds.hpp]: Same as above.

File: addons/fastroping/data/sounds/LICENSE.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Uses a sound from Tessa Elieff (Freesound.org)
```
Comment (Glowbal) [inline addons/fastroping/data/sounds/LICENSE.md]: Which sounds?

File: addons/fastroping/CfgSounds.hpp (hunk 3 above)
Comment (Glowbal) [inline addons/fastroping/CfgSounds.hpp]: Shouldn't the `PATHOF` be wrapped in a `QUOTE`?

File: addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf (hunk 1 above)
Comment (nomisum) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: feel free to modify, i'm a code noob

Comment (jonpas) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: PFH

```
[{
    params ["_args", "_idPFH"];
    _args params ["_unit"];

    if (isNull (attachedTo _unit)) exitWith {
        [_idPFH] call CBA_fnc_removePerFrameHandler;
    };

    playSound "ACE_Fastroping_Rope";
}, 1, [_unit]] call CBA_fnc_addPerFrameHandler;
```
(excuse errors, wrote it from my head)

**Comment by esteldunedain** [comment]
Guys, hold on a sec; the code and config entries are placeholders. The real content of the PR are the sounds.

He specifically said on Slack that it was not ready or tested.

File: addons/fastroping/data/sounds/LICENSE.md (hunk 4 above)
Comment (jonpas) [inline addons/fastroping/data/sounds/LICENSE.md]: This is my format of doing it, adapted from some general guidelines about licensing (https://github.com/Theseus-Aegis/tac-a3-mods/blob/master/addons/radios/sounds/LICENSES.md):
```
**Name (fileName.ogg):**  
"[Title](url)" by [author](authorUrl) is licenseType / changesHere
```

File: addons/fastroping/CfgSounds.hpp (hunk 3 above)
Comment (commy2) [inline addons/fastroping/CfgSounds.hpp]: That's what he is saying.

Comment (jonpas) [inline addons/fastroping/CfgSounds.hpp]: :balloon:

File: addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf (hunk 1 above)
Comment (commy2) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: Probably waitUntilAndExecute. The condition could play the sound.

Comment (esteldunedain) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: No need for waitUntilAndExecute. A succesion of waitAndExecute, that each check if the unit is still fastroping should be enough

Comment (jonpas) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: I think PFH is fine, it's only every 1 second and less complicated than waitAndExecute.

File: addons/fastroping/data/sounds/LICENSE.md (hunk 4 above)
Comment (nomisum) [inline addons/fastroping/data/sounds/LICENSE.md]: i'll adapt that

File: addons/fastroping/CfgSounds.hpp [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CfgSounds {
+    class ACE_Fastroping_Rope {
```
Comment (BaerMitUmlaut) [inline addons/fastroping/CfgSounds.hpp]: Could use GVAR here. Used in the rest of fastropings config.

Comment (jonpas) [inline addons/fastroping/CfgSounds.hpp]: Yeah, use `class GVAR(Rope)`.

File: addons/fastroping/CfgSounds.hpp [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+        name = "ACE_Fastroping_Rope";
+        sound[] = {QUOTE(PATHTOF(data\sounds\fastroping_rope.ogg)), db0, 1.0};
+        titles[] = {};
+    };
+    class ACE_Fastroping_Thud {
```
Comment (jonpas) [inline addons/fastroping/CfgSounds.hpp]: `class GVAR(Thud)`

File: addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf [hunk 7]
```diff
@@ -31,10 +31,23 @@ if (animationState _unit != "ACE_FastRoping") exitWith {
...
 
+[{
+    params ["_args", "_idPFH"];
+    _args params ["_unit"];
+
+    if (isNull (attachedTo _unit)) exitWith {
+        [_idPFH] call CBA_fnc_removePerFrameHandler;
+    };
+
+    playSound "ACE_Fastroping_Rope";
```
Comment (jonpas) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: `playSound QGVAR(Rope)`

File: addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf [hunk 8]
```diff
@@ -31,10 +31,23 @@ if (animationState _unit != "ACE_FastRoping") exitWith {
...
+}, 1, [_unit]] call CBA_fnc_addPerFrameHandler;
+
 //End of fast rope
 if (isNull attachedTo _unit) exitWith {
     [_unit, "", 2] call EFUNC(common,doAnimation);
     _unit setVectorUp [0, 0, 1];
 
+    playSound "ACE_Fastroping_Thud";
```
Comment (jonpas) [inline addons/fastroping/functions/fnc_fastRopeLocalPFH.sqf]: `playSound QGVAR(Thud)`

**Comment by jonpas** [comment]
@nomisum if you can resolve the comments I think this will be good for merging.

**Comment by nomisum** [comment]
@jonpas done (as far as i understood the suggestions) :smile_cat: thanks

**Comment by esteldunedain** [comment]
Ups, I think I've found a logic problem. The `fnc_fastRopeLocalPFH.sqf` function is already called each frame from a PFH. So in fact the current code would be adding a LOT of secndary PFHs playing the sound, which is not needed.

I'll a make a few modifications to the code before merging to prevent that.

**Comment by esteldunedain** [comment]
@nomisum, a simple fix, if you want to do it yourself, would be moving the code from lines 34-44 inside the exitWith of line 28, so the new PFH is only added once.

**Comment by nomisum** [comment]
will do

**Comment by jonpas** [comment]
:+1:

**Pull request merged**